src/TesApi.Web/Scheduler.cs

```diff
@@ -204,36 +214,117 @@ private async Task ProcessTesTasksContinuouslyAsync(Expression<Func<TesTask, boo
+            var storageSegments = StorageAccountUrlSegments.Create(blobUri.ToString());
+            var blobServiceClient = new BlobServiceClient(blobUri);
+            var container = blobServiceClient.GetBlobContainerClient(storageSegments.ContainerName);
+            var virtualDirectory = storageSegments.BlobName + "/";
+            var blobName = $"{virtualDirectory}{tesTaskId}.json";
+            await container.DeleteBlobIfExistsAsync(blobName, default, default, stoppingToken);
+        }
 
-            // Prioritize processing terminal state tasks, then randomize order within each state to prevent head-of-line blocking
-            foreach (var tesTask in tesTasks.OrderByDescending(t => t.State).ThenBy(t => random.Next()))
-            {
-                if (areExistingTesTasks && processNewTasksLock.CurrentCount == 0)
-                {
-                    // New tasks found, stop processing existing tasks
-                    break;
-                }
+        private async Task ProcessTesTaskCompletionsContinuouslyAsync(CancellationToken stoppingToken)
+        {
+            var sw = Stopwatch.StartNew();
 
+            while (!stoppingToken.IsCancellationRequested)
+            {
                 try
                 {
-                    await ProcessTesTask(tesTask, stoppingToken);
+                    sw.Restart();
+
+                    // TODO - this should return a container token
+                    var blobUri = new Uri(await storageAccessProvider.GetInternalTesBlobUrlAsync("task-completions", stoppingToken));
+                    var storageSegments = StorageAccountUrlSegments.Create(blobUri.ToString());
+                    var blobServiceClient = new BlobServiceClient(blobUri);
+                    var container = blobServiceClient.GetBlobContainerClient(storageSegments.ContainerName);
+                    var virtualDirectory = storageSegments.BlobName + "/";
+                    var enumerator = container.GetBlobsAsync(prefix: virtualDirectory).GetAsyncEnumerator();
+                    var completedTesTaskIds = new List<string>();
+
+                    while (await enumerator.MoveNextAsync())
+                    {
+                        // example: tasks-completions/12345678_12345678123456781234567812345678.json
+                        var blobName = enumerator.Current.Name;
+                        var json = (await container.GetBlobClient(blobName).DownloadContentAsync(stoppingToken)).Value.Content.ToString();
+                        var tesTaskCompletionMessage = JsonConvert.DeserializeObject<TesTaskCompletionMessage>(json);
+                        completedTesTaskIds.Add(tesTaskCompletionMessage.Id);
+                    }
+
+                    // Set class-level object so that it can be used by the existing task processor thread
+                    this.completedTesTaskIds = completedTesTaskIds;
```

Comment:
fyi; this is how new TES task completions are communicated to the existing task processing thread